runapp.py

```diff
@@ -23,16 +23,28 @@
 from newrelic import agent
 agent.initialize()
 
+from webob.dec import wsgify
+from webob.exc import HTTPMovedPermanently
+
 from paste.deploy import loadapp
 from pyramid.paster import setup_logging
 from raven.middleware import Sentry
 from waitress import serve
 
+
+@wsgify.middleware
+def ForceTLSMiddleware(req, app):
+    if 'X-Forwarded-Proto' in req.headers and req.headers['X-Forwarded-Proto'] != 'https':
+        https_url = req.url.replace('http://', 'https://')
```

Comment:
As part of the OAuth2 protocol it is possible to put URLs as the callback parameter in the query string and in that case this replace will be replacing too many things.

I think it's better to use the urlparse module and only change the scheme.